pkg/controller/controller_ref_manager.go

```diff
@@ -183,6 +183,10 @@ func (m *PodControllerRefManager) ClaimPods(pods []*v1.Pod, filters ...func(*v1.
 
 	match := func(obj metav1.Object) bool {
 		pod := obj.(*v1.Pod)
+		// Skip pods marked for deletion
```

Comment:
By adding this to `match()`, it means that controllers will release Pods that they own once they are marked for deletion.

I guess what you wanted is for controllers to neither adopt nor release Pods marked for deletion. The way to do that would be to filter the original `pods` list before calling `claimObject()` at all. There is no "hook" on `ClaimPods()` to do this because the caller can simply do that before passing `pods` in.

Note that this is very similar to what used to be done, and I moved it out because it broke some controllers (e.g. DaemonSet):

https://github.com/kubernetes/kubernetes/commit/01d025a7cca4ab5db5b10ab720991cb77d40cc51

For controllers that previously expected such filtering (RC/RS), I moved it to before the call to `ClaimPods()` in the above commit. If there are other controllers that also expect such filtering, I think we should add similar code there rather than forcing this on all controllers.